apps/web/src/modules/create-proposal/components/TransactionForm/Migration/Migration.tsx

```diff
@@ -0,0 +1,172 @@
+import { Stack } from '@zoralabs/zord'
+import axios from 'axios'
+import { FormikHelpers } from 'formik'
+import { useRouter } from 'next/router'
+import { ReactNode } from 'react'
+import useSWR from 'swr'
+import { encodeFunctionData, parseAbi } from 'viem'
+import { useBalance, useContractRead } from 'wagmi'
+
+import { L1_MESSENGERS, L2_DEPLOYMENT_ADDRESSES } from 'src/constants/addresses'
+import { auctionAbi, tokenAbi } from 'src/data/contract/abis'
+import { DaoMember } from 'src/data/subgraph/requests/memberSnapshot'
+import { TransactionType } from 'src/modules/create-proposal/constants/transactionType'
+import { useProposalStore } from 'src/modules/create-proposal/stores'
+import { prepareMerkle } from 'src/modules/create-proposal/utils/prepareMerkle'
+import { useDaoStore } from 'src/modules/dao/stores/useDaoStore'
+import { useChainStore } from 'src/stores/useChainStore'
+import { AddressType, CHAIN_ID } from 'src/typings'
+
+import { MigrationFormC1 } from './MigrationFormC1'
+import { MigrationFormC2 } from './MigrationFormC2'
+import { MigrationFormC2Values } from './MigrationFormC2.schema'
+import { MigrationTracker } from './MigrationTracker'
+import { prepareMigrationDeploy } from './prepareMigrationDeploy'
+
+type MembersQuery = {
+  membersList: DaoMember[]
+}
+
+export const Migration: React.FC = () => {
+  const chain = useChainStore((x) => x.chain)
+  const thisDao = useDaoStore()
+  const thisChain = useChainStore()
+  const { treasury, auction, token } = useDaoStore((x) => x.addresses)
+  const addTransaction = useProposalStore((state) => state.addTransaction)
+  const { isReady } = useRouter()
+
+  const { data: treasuryBalance } = useBalance({
+    address: treasury as `0x${string}`,
+    chainId: chain.id,
+  })
+
+  const { data: paused } = useContractRead({
+    abi: auctionAbi,
+    address: auction,
+    functionName: 'paused',
+    chainId: chain.id,
+  })
+
+  const { data: existingFounders } = useContractRead({
+    abi: tokenAbi,
+    address: token,
+    functionName: 'getFounders',
+    chainId: chain.id,
+  })
+
+  let merkleRoot: `0x${string}`
+
+  const {
+    data: members,
+    error,
+    isValidating,
+  } = useSWR(isReady ? [token, chain.id] : undefined, () =>
```

Comment:
we only need to run this query if the DAO is on step 2 right? currently this will run every time

Reply:
Yeah, I think the issue here is that I need it in handleC2Submit, which is declared at this level, because the C2Form is rendered here. Since useSWR is a hook, it can't be inside of handleC2Submit, however. Is there a way to get this without SWR  in handleC2Submit ? Or maybe I can enable the hook only if we're on Step 2?

Reply:
yea only enable the hook if we are on step 2